runtime/compiler/control/CompilationThread.cpp

```diff
@@ -9120,15 +9323,117 @@ TR::CompilationInfo::compilationEnd(J9VMThread * vmThread, TR::IlGeneratorMethod
+                           void * compressedData = comp->trMemory()->allocateHeapMemory(codeSize+dataSize);
+
+                           // Copy the metadata and codedata combined into single buffer so that we can compress it together
+                           memcpy(originalData, dataStart, dataSize);
+                           memcpy((U_8*)(originalData)+dataSize, codeStart, codeSize);
+                           int aotMethodHeaderSize = sizeof(J9JITDataCacheHeader)+sizeof(TR_AOTMethodHeader);
+                           int compressedDataSize = deflateBuffer((U_8*)(originalData)+aotMethodHeaderSize, dataSize+codeSize-aotMethodHeaderSize, (U_8*)(compressedData)+aotMethodHeaderSize, Z_DEFAULT_COMPRESSION);
+                           if (compressedDataSize != COMPRESSION_FAILED)
+                              {
+                              /**
+                               * We are going to store both metadata and code in contiguous memory in shared class cache.
+                               * In load run we get the whole buffer from the cache and use the information from header to
+                               * Get the relocation data and compiled code.
+                               * Because of this reason we copy both metadata and compiled code in one buffer and deflate it together.
+                               * TODO: We will always query the shared class cache to get full data for stored AOT compiled method 
+                               * that is combined meta data and code data. Even if compression of AOT bytes is disabled, we do not need
+                               * to send two different buffers to store in cache and also as no one queries either code data / metadata for
+                               * method, clean up the share classs cache API.
+                               */
+                              aotMethodHeaderEntry->flags |= TR_AOTMethodHeader_CompressedMethodInCache;
+                              memcpy(compressedData, dataStart, aotMethodHeaderSize);
+                              metadataToStore = (const U_8*) compressedData;
+                              metadataToStoreSize = compressedDataSize+aotMethodHeaderSize;
+                              codedataToStore = NULL;
+                              codedataToStoreSize = 0;  
+                              if (TR::Options::getVerboseOption(TR_VerboseAOTCompression))
+                                 {
+                                 J9UTF8 *className;
+                                 J9UTF8 *name;
+                                 J9UTF8 *signature;
+                                 getClassNameSignatureFromMethod(method, className, name, signature);
+                                 TR_VerboseLog::writeLineLocked(TR_Vlog_AOTCOMPRESSION, "%.*s.%.*s%.*s : Compression of method data Successful - Original method size = %d bytes, Compressed Method Size = %d bytes",
```

Comment:
I believe you have access to the comp object which can return the entire name of the method being compiled as comp->signature(). This may be simpler than printing the parts of the method individually.

Reply:
Changed in https://github.com/eclipse/openj9/pull/5926/commits/66c4368333f9f5ba388b64ec16893d59d9c7eaa1